dvc/repo/fetch.py

```diff
@@ -82,18 +101,20 @@ def _fetch_external(self, repo_url, repo_rev, files, jobs):
             if is_dvc_repo:
                 repo.cache.local.cache_dir = self.cache.local.cache_dir
                 with repo.state:
-                    cache = NamedCache()
+                    used_cache = []
```

Comment:
From the old PR:
> Hm, not sure I understand why this is needed.

`Output.get_used_cache()` now returns a tuple of `(used_dir_cache, used_file_cache)`, so we collect a list of these tuples instead of merging all of the used dir/file cache entries into a single `NamedCache`